src/sentry/shared_integrations/client/base.py

```diff
@@ -464,3 +468,29 @@ def disable_integration(self, buffer) -> None:
                 data={"provider": rpc_integration.provider},
             )
         return
+
+    def disable_plugin(self, buffer) -> None:
+        plugin = plugins.get(self.slug)
+        if not plugin.is_enabled():
+            return
+
+        project = Project.objects.get(id=self.project_id)
+
+        extra = {
+            "project_id": self.project_id,
+            "buffer_record": buffer._get_all_from_buffer(),
+            "provider": self.plugin_name,
+        }
+        if project:
+            extra["organization_id"] = project.organization.id
+        else:
+            extra["organization_id"] = "unknown"
+
+        self.logger.info(
+            "plugin.disabled",
+            extra=extra,
+        )
+        if features.has("organizations:plugin-disable-on-broken", project.organization):
+            plugin.disable()
+            notify_disable(project.organization, self.plugin_name, self._get_redis_key())
+            buffer.clear()
```

Comment:
We should have tests for this path, but totally understandable if you don't have time today to finish :p I can pick it up later

Reply:
Added some tests for trello!